lib/pf/Portal/ProfileFactory.pm

```diff
@@ -33,6 +35,8 @@ Create a new pf::Portal::Profile instance based on parameters given.
 
 =cut
 
+tie our $PROFILE_FILTER_ENGINE , 'pfconfig::cached_scalar' => 'resource::ProfileFilterEngine';
```

Comment:
The filter engine(s) should be under either filter_engine::profile or resource::filter_engine::profile
I personally prefer the first but they shouldn't be mixed with normal resources as there will be a few of them